### ISSUE TITLE
example attribute in Field is obsolete in JSON schema and fails to pass mypy check

### ISSUE THREAD

**Issue by mshonichev** [koxudaxi/datamodel-code-generator#1721]
**Describe the bug**
TL;DR: `datamodel-codegen` generates pydantic models with obsolete missing `example` field, which in turn fails static type checking.

**To Reproduce**

Example schema:
```json
{
    "swagger": "2.0",
    "info": {
        "description": "Swagger API description for app.",
        "title": "App API",
        "contact": {}
    },
    "paths": {
        "/v1/obj": {
            "post": {
                "description": "Create object",
                "consumes": [
                    "application/json"
                ],
                "produces": [
                    "application/json"
                ],
                "tags": [
                    "Version v1.0.0"
                ],
                "summary": "Create obj",
                "parameters": [
                    {
                        "description": "New obj parameters",
                        "name": "data",
                        "in": "body",
                        "required": true,
                        "schema": {
                            "$ref": "#/definitions/api.CreateObjRequest"
                        }
                    }
                ],
                "responses": {
                    "202": {
                        "description": "Accepted",
                        "schema": {
                            "$ref": "#/definitions/api.CreateObjResponse"
                        }
                    },
                    "500": {
                        "description": "Internal Server Error",
                        "schema": {
                            "$ref": "#/definitions/api.InternalServerError"
                        }
                    }
                }
            }
        },
    },
    "definitions": {
        "api.CreateObjRequest": {
            "type": "object",
            "required": [
                "name",
                "data"
            ],
            "properties": {
                "name": {
                    "description": "Name of object",
                    "type": "string",
                    "maxLength": 200,
                    "example": "my object"
                },
                "data": {
                    "description": "object opaque data",
                    "type": "string",
                    "maxLength": 200
                }
            }
        },
        "api.CreateObjResponse": {
            "type": "object",
            "properties": {
                "id": {
                    "description": "obj id",
                    "type": "string",
                    "example": "2ef01247-936d-4855-a3e4-6d5b024fcf7d"
                }
            }
        },
        "api.InternalServerError": {
            "description": "describes error occurred on server because something went wrong",
            "type": "object",
            "properties": {
                "code": {
                    "description": "Describes what happened. 5 - common server error",
                    "type": "integer",
                    "example": 5
                },
                "message": {
                    "description": "Error message. May vary depends on what happened",
                    "type": "string",
                    "example": "Internal server error"
                }
            }
        }
    }
}

```

Used commandline:
```
$ datamodel-codegen --input-file-type jsonschema --output ./models --output-model-type pydantic_v2.BaseModel --use-field-description --use-standard-collections --use-union-operator --use-unique-items-as-set --target-python-version 3.11  --use-schema-description  --use-double-quotes  --reuse-model --disable-warnings --keep-model-order --enable-version-header --field-constraints --use-default-kwarg --remove-special-field-name-prefix --enum-field-as-literal all
```

**Expected behaviour**
A model generated must be valid, e.g. don't use obsolete attributes and pass static type checking and listing .

**Actual behaviour**
Following model is generated for `api.CreateObjRequest`:

```python
class CreateObjRequest(BaseModel):
    name: str = Field(..., example="my object", max_length=200)
    """
    Name of object
    """
    data: str
    """
    Opaque object data
    """
```

However, checking this model with `mypy` reveals following bug:
```
$ mypy ./models/*.py

/home/src/.venv/lib/python3.11/site-packages/pydantic/fields.py:672: note: "Field" defined here
src/models/api.py:636: error: Unexpected keyword argument "example" for "Field"; did you mean "examples"?  [call-arg]
```

Turns out that `example` attribute was deprecated and removed long time ago. 

There's a very long article describing reasons and current situation with the standards.
https://fastapi.tiangolo.com/tutorial/schema-extra-example/

As a user, IMHO best variant to fix this issue until standards inconsistencies stabilize would be to implement an option  similar to https://github.com/koxudaxi/datamodel-code-generator/issues/1690 
which when enabled orders generator to format this `example` as a part of object attribute docstring, e.g.

```python
class CreateObjRequest(BaseModel):
    name: str = Field(..., max_length=200)
    """
    Name of object

    Example: "my object"
    """
    data: str
    """
    Opaque object data
    """
```

Another option is to simply remove this attribute or automagically convert it from `example` to list of `examples`, as standards seem to suggest.

**Version:**
 - Python version: 3.10
 - datamodel-code-generator version: [e.g. 0.22.1]

**Additional context**

```
$ pip freeze | grep -Ei "datamod|pydan|mypy"
datamodel-code-generator==0.22.1
mypy==1.6.1
mypy-extensions==1.0.0
pydantic==2.4.2
pydantic-yaml==1.2.0
pydantic_core==2.10.1
```

**Comment by koxudaxi** [comment]
@mshonichev 
Thank you for creating the issue.
The issue is fixed in the latest version.
